curriculum/challenges/english/25-front-end-development/lecture-understanding-form-validation/6733aae9d25004f60d1e86f2.md

```diff
@@ -10,6 +10,142 @@ dashedName: what-are-some-ways-to-validate-forms-using-javascript
 
 Watch the video lecture and answer the questions below.
 
+# --transcript--
+
+What are some ways to validate forms using JavaScript?
+
+Let’s learn about how to use JavaScript to validate your forms.
+
+In a previous lecture, you’ve learned how to use HTML to restrict the values your users can submit in your form. But sometimes that’s not enough. If you want to get more complex, such as displaying your own error messages to the user, you will need to use JavaScript.
+
+Certain HTML elements, such as the `textarea` and `input` elements, expose a constraint validation API. This API allows you to assert that the user’s provided value for that element passes any HTML-level validation you have written, such as minimum length or pattern matching.
+
+But how can you actually use it?
+
+Let’s say you wanted employees at a company to send feedback messages through a form like this:
+
+```html
+<form>
+  <label>Enter your email: </label>
+  <input required type="email" />
+
+  <label>Enter your feedback: </label>
+  <textarea required placeholder="Your feedback here..."></textarea>
+
+  <button type="submit">Submit Feedback</button>
+</form>
+```
+
+We are using the email input which comes with built in validation to check for basic validation like if the input includes the at(`@`) sign.
+
+But what if the user provides an email address like `example@email.com`?
+
+This would pass the basic validation, but we want to be more specific about accepting emails from those with a company email address.
+
+This is where we can use the `pattern` attribute to specify that the email address must end in a company email address.
+
+Here is what the updated example will look like:
+
+```html
+<form>
+  <label>Enter your email: </label>
+  <input required placeholder="username@sampleCompany.com" type="email" pattern=".+@sampleCompany\.com" />
+
+  <label>Enter your feedback: </label>
+  <textarea required placeholder="Your feedback here..."></textarea>
+
+  <button type="submit">Submit Feedback</button>
+</form>
+```
+
+Now, if you try to submit the feedback, you will see a message saying "Please match the requested format."
+
+Even though the input does have placeholder text showing them the desired format, it would be better to also include a customized error message using JavaScript.
```

Comment:
```suggestion
Even though the `input` does have placeholder text showing them the desired format, it would be better to also include a customized error message using JavaScript.
```